app/src/main/java/com/uc/mymovies/MainActivity.kt

```diff
@@ -5,25 +5,27 @@ import android.os.Bundle
 import android.util.Log
 import android.widget.ListAdapter
 import android.widget.ListView
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelProviders
 import com.google.firebase.firestore.FirebaseFirestore
 import com.google.firebase.firestore.FirebaseFirestoreSettings
+import com.uc.mymovies.main.MainFragment
+import com.uc.mymovies.main.MainViewModel
 import com.uc.mymovies.service.MovieService
 
 class MainActivity : AppCompatActivity() {
-
-    var movieService: MovieService = MovieService()
-
-    init {
-        fetchMovies()
-    }
-
-    private fun fetchMovies() {
-       var result = movieService.fetchUserMovies("connorkeith")
-        Log.d("TEST-LOG: ", result.toString())
-    }
+    // create fragment for each layout in a new package
+     private lateinit var mainFragment: MainFragment
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+       // mainFragment = MainFragment.newInstance()
+        val viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
+        savedInstanceState?.let {
+            supportFragmentManager.beginTransaction()
+                .replace(R.id.container, mainFragment)
+                .commitNow()
+        }
```

Comment:
use let function to avoid making explicit null validations of the variable
https://blog.intive-fdv.com/best-practices-to-write-a-clean-code-with-kotlin-part-ii/

Reply:
Yes, good idea.